src/main/java/co/rsk/federate/signing/hsm/message/ParsedHeader.java

```diff
@@ -0,0 +1,32 @@
+package co.rsk.federate.signing.hsm.message;
+
+import org.ethereum.core.BlockHeader;
+import org.spongycastle.util.encoders.Hex;
+
+import java.util.List;
+
+public class ParsedHeader {
+    private final String blockHeader;
+    private final String[] brothers;
+
+    public ParsedHeader(BlockHeader blockHeader, List<BlockHeader> brothers) {
+        this.blockHeader = parseBlockHeader(blockHeader);
+        this.brothers = parseBrothers(brothers);
+    }
+
+    public String getBlockHeader() {
+        return blockHeader;
+    }
+
+    public String[] getBrothers() {
+        return brothers.clone();
+    }
+
+    private String parseBlockHeader(BlockHeader blockHeader) {
```

Comment:
Being nit picky here, but I find it a bit confusing that both ParsedHeader and AdvanceBlockchainMessage classes have `parse` methods. It confuses me a bit, don't know where the parsing is really happening.

Maybe call these methods `serialize` instead of `parse`? What do yo think?

Reply:
Make Sense. Updated